src/artifact_providers/github.ts

```diff
@@ -95,20 +95,15 @@ export class GithubArtifactProvider extends BaseArtifactProvider {
 
     const allArtifacts = artifactResponse.artifacts;
 
-    // We need to find the archive where name maches the revision
-    const foundArtifacts = allArtifacts.filter(
-      artifact => artifact.name === revision
+    // We need to find the most recent archive where name matches the revision.
+    const foundArtifact = allArtifacts.reduce((result, artifact) =>
+      artifact.name === revision && result?.created_at < artifact.created_at
+        ? artifact
+        : result
```

Comment:
I think this selects `result` if `result` has no `created_at`. Is that what we want? 🤔

Reply:
Good catch, that said `result` will always have a `created_at`. This code says "if result itself is missing" which essentially boils down to the same conclusion you came to. And no, that's not something we want. I was saved here by how `reduce` works: if you don't provide an explicit starting value, it uses the first item in the array so `result` is never undefined.

Will fix this.

Reply:
Fixed.